iceoryx_utils/include/iceoryx_utils/internal/cxx/forward_list.inl

```diff
@@ -0,0 +1,538 @@
+{
+    iterator iter = before_begin();
+    return ++iter;
+}
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::const_iterator forward_list<T, CAPACITY>::begin() const noexcept
+{
+    return cbegin();
+}
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::const_iterator forward_list<T, CAPACITY>::cbegin() const noexcept
+{
+    const_iterator citer = cbefore_begin();
+    return ++citer;
+}
+
+
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::iterator forward_list<T, CAPACITY>::end() noexcept
+{
+    return iterator{this, INVALID_INDEX};
+}
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::const_iterator forward_list<T, CAPACITY>::end() const noexcept
+{
+    return cend();
+}
+template <typename T, int64_t CAPACITY>
+typename forward_list<T, CAPACITY>::const_iterator forward_list<T, CAPACITY>::cend() const noexcept
+{
+    return const_iterator{this, INVALID_INDEX};
+}
+
+
+template <typename T, int64_t CAPACITY>
+inline bool forward_list<T, CAPACITY>::empty() const noexcept
+{
+    return (m_size == 0);
+}
+
+template <typename T, int64_t CAPACITY>
+inline bool forward_list<T, CAPACITY>::full() const noexcept
+{
+    return (m_size >= CAPACITY);
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::sizeType forward_list<T, CAPACITY>::size() const noexcept
+{
+    return m_size;
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::sizeType forward_list<T, CAPACITY>::capacity() const noexcept
+{
+    return CAPACITY;
+}
+
+template <typename T, int64_t CAPACITY>
+inline typename forward_list<T, CAPACITY>::sizeType forward_list<T, CAPACITY>::max_size() const noexcept
+{
+    return capacity();
+}
+
+
+template <typename T, int64_t CAPACITY>
+template <typename... ConstructorArgs>
+bool forward_list<T, CAPACITY>::emplace_front(ConstructorArgs&&... args) noexcept
+{
+    return (cend() != emplace_after(before_begin(), std::forward<ConstructorArgs>(args)...));
+}
+
+template <typename T, int64_t CAPACITY>
+template <typename... ConstructorArgs>
+typename forward_list<T, CAPACITY>::iterator
+forward_list<T, CAPACITY>::emplace_after(const_iterator afterToBeEmplacedIter, ConstructorArgs&&... args) noexcept
+{
+    if (this != afterToBeEmplacedIter.m_cList)
+    {
+        errorMessage(__PRETTY_FUNCTION__, " iterator of other list can't be used ");
+        std::terminate();
+    }
+
+    const sizeType l_toBeUsedIdx = getHeadFreeEl()->nextIdx;
+
+    if (m_size < CAPACITY && l_toBeUsedIdx < INVALID_INDEX)
+    {
+        // unlink from freeList
+        getHeadFreeEl()->nextIdx = getPointerToListNode()[l_toBeUsedIdx].nextIdx;
+
+        // data class c'tor
+        new (&(getPointerToListNode()[l_toBeUsedIdx].data)) T(std::forward<ConstructorArgs>(args)...);
+
+        // add to usedList
+        const sizeType emplaceAfterIdx = afterToBeEmplacedIter.m_iterListNodeIdx;
```

Comment:
The variables again here are a bit confusing ...
I would consider revising these.
What does `emplaceAfterIdx` mean ?

Reply:
Idx is fine to use, it's pretty standard. The new name is good.